log/provider.go

```diff
@@ -22,5 +25,16 @@ type LoggerProvider interface {
 	//
 	// Implementations of this method need to be safe for a user to call
 	// concurrently.
-	Logger(name string, options ...LoggerOption) Logger
+	Logger(cfg LoggerConfig) Logger
+}
+
+// LoggerConfig contains options for a [Logger].
+type LoggerConfig struct {
+	// Ensure forward compatibility by explicitly making this not comparable.
+	noCmp [0]func() //nolint: unused  // This is indeed used.
+
+	Name       string
+	Version    string
+	SchemaURL  string
+	Attributes attribute.Set
```

Comment:
An almost identical alternative would be:

```suggestion
	Logger(name string, cfg LoggerConfig) Logger
}

// LoggerConfig contains options for a [Logger].
type LoggerConfig struct {
	// Ensure forward compatibility by explicitly making this not comparable.
	noCmp [0]func() //nolint: unused  // This is indeed used.

	Version    string
	SchemaURL  string
	Attributes attribute.Set
```